dom.go

```diff
@@ -169,8 +173,14 @@ func (h *HTML) Restore(old ComponentOrHTML) {
 	}
 
 	if prev, ok := old.(*HTML); ok && prev != nil {
-		h.restoreHTML(prev)
-		return
+		if h.text != "" && prev.text != "" {
+			h.restoreText(prev)
+			return
+		}
+		if h.tag != "" && prev.tag != "" && h.tag == prev.tag {
+			h.restoreHTML(prev)
+			return
+		}
```

Comment:
If both `h` and `prev` are text nodes containing an empty string, we probably shouldn't call `restoreHTML`... Not sure what you think of empty string text nodes, but they're possible to create with `vecty.Text("")`, so we should watch out for them (see https://github.com/gopherjs/vecty/pull/67)